libblkid/src/superblocks/cs_fvault2.c

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (C) 2022 Milan Broz <gmazyland@gmail.com>
+ *
+ * This file may be redistributed under the terms of the
+ * GNU Lesser General Public License.
+ */
+
+#include "superblocks.h"
+#include "crc32c.h"
+
+/*
+ * For header details, see:
+ * https://github.com/libyal/libfvde/blob/main/documentation/FileVault%20Drive%20Encryption%20(FVDE).asciidoc
+ * https://is.muni.cz/auth/th/p0aok/thesis.pdf
+ */
+
+/* Apple Core Storage magic bytes */
+#define CS_MAGIC	"CS"
+
+struct crc32_checksum {
+	uint32_t value;
+	uint32_t seed;
+} __attribute__((packed));
+
+struct cs_fvault2_sb {
+	struct crc32_checksum checksum;
+	uint16_t version;
+	uint16_t block_type;
+	uint8_t unknown1[52];
+	uint64_t ph_vol_size;
+	uint8_t unknown2[16];
+	uint16_t magic;
+	uint32_t checksum_algo;
+	uint8_t unknown3[2];
+	uint32_t block_size;
```

Comment:
Could also be reported

Reply:
I would better avoid this - Core Storage is kind of LVM layers and here we parse physical volume only; there are other two layers above that can have other metadata. Block size here should reference metadata block offsets, not block_size as encryption block size. (IIRC)

Reply:
I have added some comment to sb definition.